components/service/glean/docs/metrics/adding-new-metrics.md

```diff
@@ -108,3 +109,54 @@ assertEquals(6, Controls.refreshPressed.testGetValue())
 
 NOTE: Using the testing API requires calling `Glean.enableTestingMode()` first,
 such as from a `@Before` method.
+
+## Timing Distributions
+
+Used to accumulate and store time measurement samples into buckets, which are exponentially sized 
+using a natural log function. 
+
+If you wanted to create a timing distribution to measure page load times, first you need to add 
+an entry for the counter to the `metrics.yaml` file:
+
+```YAML
+pages:
+  page_load:
+    type: timing_distribution
+    time_unit: millisecond
+    description: >
+      Counts how long each page takes to load into exponential buckets
+    ...
+```
+
+Note that `time_unit` is required in the `metrics.yaml` file to identify the base unit of time
+used to indicate the size of the buckets and the units intended when accumulating and recording
+samples.
+
+Now you can use the event from the applications code:
+```Kotlin
+import org.mozilla.yourApplication.GleanMetrics.Pages
+
+// ...
+pages.pageLoad.accumulate(1L) // Accumulates a sample of 1 millisecond
+pages.pageLoad.accumulate(10L) // Accumulates a sample of 10 milliseconds
+```
+
+At this point, the metric would have a `sum` of `11` and a `values` map of `{ "0" : 1, "8" : 1}`
+
+There are test APIs available too:
+```Kotlin
+import org.mozilla.yourApplication.GleanMetrics.Pages
+
+// Was anything recorded?
+assertTrue(pages.pageLoad.testHasValue())
+// Does the sum have the expected value?
+val tdSnapshot = pages.pageLoad.testGetValue()
+assertEquals(11, tdSnapshot?.sum)
+assertEquals(1L, tdSnapshot?.values["0"]) // The bucket the '1L' was accumulated to
+assertEquals(1L, tdSnapshot?.values["8"]) // The bucket the '10L' was accumulated to
+assertNull(tdSnapshot?.values["1"]) // Should be null since nothing was accumulated to this bucket
```

Comment:
I think we should avoid test author having to look up the buckets.
E.g. for timings in product, it's hard to test exact values. Instead just checking *N values were recorded* seems a test pattern engineers use.
From [this search](https://searchfox.org/mozilla-central/search?q=gethistogram.%2Bms&case=false&regexp=true&path=test) i found only `sum` and `count` checks:
- [example 1](https://searchfox.org/mozilla-central/rev/7c20ad925005fbad7b8e08813115f1ec7fa1c248/browser/base/content/test/tabs/browser_tabCloseProbes.js#79)
- [example 2](https://searchfox.org/mozilla-central/rev/7c20ad925005fbad7b8e08813115f1ec7fa1c248/dom/webauthn/tests/browser/browser_webauthn_telemetry.js#150)).
- ... etc.

So how about just exposing a `testValueCount`/`testSampleCount` or so?
And just using that for the example?

Reply:
I can find more examples [with this search](https://searchfox.org/mozilla-central/search?q=_MS%22&case=true&regexp=true&path=test), but the previous pattern holds.